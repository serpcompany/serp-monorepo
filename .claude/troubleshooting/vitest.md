https://vitest.dev/guide/common-errors.html