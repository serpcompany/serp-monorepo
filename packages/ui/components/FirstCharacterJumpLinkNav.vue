<template>
  <nav class="mx-auto flex justify-between overflow-x-scroll">
    <u-button
      v-for="character in characters"
      :key="character"
      :disabled="!filteredCharacters.includes(character)"
      :to="`#${character}`"
      class="h-10 w-10 items-center bg-none text-2xl lg:justify-center"
    >
      {{ character }}
    </u-button>
  </nav>
</template>

<script setup lang="ts">
defineProps<{
  characters: string[];
  filteredCharacters: string[];
}>();
</script>
