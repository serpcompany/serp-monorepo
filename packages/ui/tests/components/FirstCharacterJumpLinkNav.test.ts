import { describe, expect, it } from 'vitest'
import FirstCharacterJumpLinkNav from '../../components/FirstCharacterJumpLinkNav.vue'
import ComponentRender from '../componentRender'

describe('firstCharacterJumpLinkNav Snapshot', () => {
  it.each([
    [
      'with no characters',
      { props: { characters: [], filteredCharacters: [] } },
    ],
    [
      'with all buttons enabled',
      {
        props: {
          characters: ['A', 'B', 'C'],
          filteredCharacters: ['A', 'B', 'C'],
        },
      },
    ],
    [
      'with some buttons disabled',
      {
        props: {
          characters: ['A', 'B', 'C'],
          filteredCharacters: ['A', 'C'],
        },
      },
    ],
    [
      'with all buttons disabled',
      {
        props: {
          characters: ['A', 'B', 'C'],
          filteredCharacters: [],
        },
      },
    ],
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(
        `FirstCharacterJumpLinkNav ${desc}`,
        options,
        FirstCharacterJumpLinkNav,
      )
      expect(html).toMatchSnapshot()
    },
  )
})
