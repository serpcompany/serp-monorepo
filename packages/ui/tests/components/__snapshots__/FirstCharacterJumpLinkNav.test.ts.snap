// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstCharacterJumpLinkNav Snapshot > renders with all buttons disabled correctly 1`] = `
<nav class="
  flex
  justify-between
  mx-auto
  overflow-x-scroll
">
  <a
    aria-disabled="true"
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    role="link"
    tabindex="-1"
  >
    
  <!-- v-if -->A
    
  <!-- v-if -->
  </a>
  <a
    aria-disabled="true"
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    role="link"
    tabindex="-1"
  >
    
  <!-- v-if -->B
    
  <!-- v-if -->
  </a>
  <a
    aria-disabled="true"
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    role="link"
    tabindex="-1"
  >
    
  <!-- v-if -->C
    
  <!-- v-if -->
  </a>
</nav>
`;

exports[`FirstCharacterJumpLinkNav Snapshot > renders with all buttons enabled correctly 1`] = `
<nav class="
  flex
  justify-between
  mx-auto
  overflow-x-scroll
">
  <a
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    href="#A"
  >
    
  <!-- v-if -->A
    
  <!-- v-if -->
  </a>
  <a
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    href="#B"
  >
    
  <!-- v-if -->B
    
  <!-- v-if -->
  </a>
  <a
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    href="#C"
  >
    
  <!-- v-if -->C
    
  <!-- v-if -->
  </a>
</nav>
`;

exports[`FirstCharacterJumpLinkNav Snapshot > renders with no characters correctly 1`] = `
<nav class="
  flex
  justify-between
  mx-auto
  overflow-x-scroll
"></nav>
`;

exports[`FirstCharacterJumpLinkNav Snapshot > renders with some buttons disabled correctly 1`] = `
<nav class="
  flex
  justify-between
  mx-auto
  overflow-x-scroll
">
  <a
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    href="#A"
  >
    
  <!-- v-if -->A
    
  <!-- v-if -->
  </a>
  <a
    aria-disabled="true"
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    role="link"
    tabindex="-1"
  >
    
  <!-- v-if -->B
    
  <!-- v-if -->
  </a>
  <a
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    href="#C"
  >
    
  <!-- v-if -->C
    
  <!-- v-if -->
  </a>
</nav>
`;

exports[`firstCharacterJumpLinkNav Snapshot > renders with all buttons disabled correctly 1`] = `
<nav class="
  flex
  justify-between
  mx-auto
  overflow-x-scroll
">
  <a
    aria-disabled="true"
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    role="link"
    tabindex="-1"
  >
    
  <!-- v-if -->A
    
  <!-- v-if -->
  </a>
  <a
    aria-disabled="true"
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    role="link"
    tabindex="-1"
  >
    
  <!-- v-if -->B
    
  <!-- v-if -->
  </a>
  <a
    aria-disabled="true"
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    role="link"
    tabindex="-1"
  >
    
  <!-- v-if -->C
    
  <!-- v-if -->
  </a>
</nav>
`;

exports[`firstCharacterJumpLinkNav Snapshot > renders with all buttons enabled correctly 1`] = `
<nav class="
  flex
  justify-between
  mx-auto
  overflow-x-scroll
">
  <a
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    href="#A"
  >
    
  <!-- v-if -->A
    
  <!-- v-if -->
  </a>
  <a
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    href="#B"
  >
    
  <!-- v-if -->B
    
  <!-- v-if -->
  </a>
  <a
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    href="#C"
  >
    
  <!-- v-if -->C
    
  <!-- v-if -->
  </a>
</nav>
`;

exports[`firstCharacterJumpLinkNav Snapshot > renders with no characters correctly 1`] = `
<nav class="
  flex
  justify-between
  mx-auto
  overflow-x-scroll
"></nav>
`;

exports[`firstCharacterJumpLinkNav Snapshot > renders with some buttons disabled correctly 1`] = `
<nav class="
  flex
  justify-between
  mx-auto
  overflow-x-scroll
">
  <a
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    href="#A"
  >
    
  <!-- v-if -->A
    
  <!-- v-if -->
  </a>
  <a
    aria-disabled="true"
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    role="link"
    tabindex="-1"
  >
    
  <!-- v-if -->B
    
  <!-- v-if -->
  </a>
  <a
    class="
      aria-disabled:bg-primary
      aria-disabled:cursor-not-allowed
      aria-disabled:opacity-75
      bg-none
      bg-primary
      disabled:bg-primary
      disabled:cursor-not-allowed
      disabled:opacity-75
      focus-visible:outline-2
      focus-visible:outline-offset-2
      focus-visible:outline-primary
      font-medium
      gap-1.5
      h-10
      hover:bg-primary/75
      inline-flex
      items-center
      lg:justify-center
      px-2.5
      py-1.5
      rounded-md
      text-2xl
      text-inverted
      transition-colors
      w-10
    "
    href="#C"
  >
    
  <!-- v-if -->C
    
  <!-- v-if -->
  </a>
</nav>
`;
